RELEASE/scripts/autoscend/auto_boris.ash

```diff
@@ -275,13 +275,73 @@ void boris_buySkills()
 	set_property("auto_borisSkills", my_level());
 }
 
+void borisDemandSandwich()
+{
+	//Boris can summon a sandwich 3 times per day at cost of 5 MP.
+	if(!in_boris())
+	{
+		return;
+	}
+	if(get_property("_demandSandwich").to_int() > 2) 		//max 3 casts a day
+	{
+		return;
+	}
+	
+	//use ongoing MP recovery to summon sandwiches as boris if you can get the best sandwich.
+	if(my_level() > 8)
+	{
+		while(my_mp() > 4 && get_property("_demandSandwich").to_int() < 3)
+		{
+			use_skill(1, $skill[Demand Sandwich]);
+		}
+	}
+	//if your level is too low for the best sandwich, summon a single inferior sandwich to tide you over when low on adventures and if you don't already have one in inventory.
+	//this part semi relies on a future update to consumption code to make it eat one item at a time.
+	else if(my_adventures() < 8 && item_amount($item[club sandwich]) == 0 && item_amount($item[PB&BP]) == 0)
+	{
+		if(my_mp() > 4 && get_property("_demandSandwich").to_int() < 3)
+		{
+			use_skill(1, $skill[Demand Sandwich]);
+		}
+	}
+}
+
+void borisOngoingLaugh()
+{
+	//ongoing conversion of MP to HP to be done after every adventure to prevent wasting MP regen from clancy.
+	//Laugh it off costs 1 MP to cast and gives either 1 or 2 HP randomly.
+	if(!in_boris())
+	{
+		return;
+	}
+	
+	//don't waste HP on convert, keep at least 10 MP and at least 50% of MP.
+	while((my_hp()+1) < my_maxhp() && my_mp() > 10 && my_mp() > (my_maxmp()/2))
```

Comment:
`(my_hp() + 1) < my_maxhp()` -> `my_hp() <= my_maxhp()`

Reply:
We do not want <=, if HP is at 300/300 then <= is true, we do not want to try to heal here.
We only want to try and heal if HP is at least 2 under. that is:
298/300 heal
299/300 do not heal, 50% chance of wasting 1 HP aka we are wasting half an MP healing now
300/300 will just error because you cannot heal when at 100%.

If you want it to be <= then it needs to change `(my_hp() + 1) < my_maxhp()` into 
`(my_hp()+2) <= my_maxhp()`

Reply:
Alright, I redid this whole thing to be based on actual predicted wasted MP instead of just spending MP down to 50%